analytics/models.py

```diff
@@ -1,5 +1,26 @@
+from django.conf import settings
 from django.db import models
+from django.dispatch import receiver
+from . import authentication
+
+# Create your models here.
+
+
+# Note: authentication-related models are defined in authentication.py.
+
+
+@receiver(models.signals.post_save, sender=settings.AUTH_USER_MODEL)
+def create_auth_token(sender, instance=None, created=False, **_kwargs):
+    """Provide tokens to users automatically."""
+    if created:
+        token = authentication.AnalyticsToken.objects.create(user=instance)
+        token.save()
 
 
 class Event(models.Model):
-    pass
+    name = models.CharField(max_length=255)
+    created_at = models.DateTimeField(auto_now_add=True)
+    additional_data = models.TextField(blank=True, default='')
```

Comment:
w takim wypadku faktycznie mamy opcjonalny tekst jednak w bazie pole nie jest nullowalne, zmieniłbym na => `additional_data = models.TextField(blank=True, null=True)`

Reply:
@Jeziersky nooo, tu się nie zgodzę, `default=''` też pasi

Reply:
@Jeziersky faktycznie, odniosłem "optional text" do procesu tworzenia eventu, ale teraz widzę, że mogło chodzić o opcjonalność w sensie języka typów. Technicznie pusty napis jest prostszy, ale takie rozumienie jest elegantsze - skoro zmiana jest sugerowana, to jasne. cb1546ce393638d729bc5df9b811586abb7c55a7 wprowadza zmianę. f624fd4914842d0ffe8329d7ec24d3a99e5df91e poprawia test, żeby pasował do zmienionego założenia.